### PR TITLE
[web] specify all defines used for html, ck, auto rendering modes

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Allow changing the default value of the flutter web rendering backends in debug mode by specifying all dart defines. Also updates the BUILD.gn comments to be clearer on what is getting built.

### CODE REVIEW

**Review by angjieli — APPROVED** [review]
LGTM

File: web_sdk/BUILD.gn [hunk 1]
```diff
@@ -286,6 +289,7 @@ prebuilt_dart_action("flutter_dartdevc_canvaskit_html_kernel_sdk") {
     "--libraries-file",
     "org-dartlang-sdk:///flutter_web_sdk/libraries.json",
     "--inline-source-map",
+    "-DFLUTTER_WEB_USE_SKIA=true",
```
Comment (jonahwilliams) [inline web_sdk/BUILD.gn:292]: Does autodetect use skia=true?

Comment (angjieli) [reply web_sdk/BUILD.gn:292]: You can remove this. It doesn't matter once FLUTTER_WEB_AUTO_DETECT is specified.

Comment (ferhatb) [reply web_sdk/BUILD.gn:292]: autodetect has higher precedence, it doesn't check skia at that point.

**Review by ferhatb — APPROVED** [review]

File: web_sdk/BUILD.gn (hunk 1 above)
Comment (jonahwilliams) [reply web_sdk/BUILD.gn:292]: OK, I'll leave it as set just so it is easier to change in the future.

**Pull request merged**